tests/integration/helpers.py

```diff
@@ -72,8 +111,98 @@ async def run_mongo_op(ops_test: OpsTest, mongo_op: str):
     if ret_code != 0:
         logger.error("code %r; stdout %r; stderr: %r", ret_code, stdout, stderr)
+        output.failed = True
+    else:
+        output.succeeded = True
+        if expecting_output:
+            try:
+                output.data = json.loads(stdout)
+            except Exception as e:
+                logger.error(f"Could not serialize the output into json: \n{stdout}")
+                raise e
+
+    return output
+
+
+def primary_host(rs_status_data) -> Optional[str]:
+    """Returns the primary host in the replica set or None if none was elected."""
+    primary_list = [
+        member["name"]
+        for member in rs_status_data["members"]
+        if member["stateStr"].upper() == "PRIMARY"
+    ]
+
+    if not primary_list:
         return None
-    return json.loads(stdout)
+
+    return primary_list[0]
+
+
+async def check_if_test_documents_stored(
+    ops_test: OpsTest, collection: str, mongo_uri: str = None
+) -> None:
+    # decide whether to pass a mongo_uri or replication set to the "run_mongo_op" function
+    run_mongo_op_kwargs = {"suffix": f"?replicaSet={APP_NAME}"}
+    if mongo_uri is not None:
+        run_mongo_op_kwargs["mongo_uri"] = mongo_uri
+
+    # serialize the str test documents into json
+    o_test_docs = json.loads(TEST_DOCUMENTS)
+
+    # query filter
+    query_filter = json.dumps({"$or": [{"uid": test_doc["uid"]} for test_doc in o_test_docs]})
+
+    count_documents = await run_mongo_op(
+        ops_test, f"db.{collection}.countDocuments({query_filter})", **run_mongo_op_kwargs
+    )
+    assert count_documents.succeeded and count_documents.data == 2
+
+    # descending order to match insertion order of the test documents
+    find_documents = await run_mongo_op(
+        ops_test,
+        f"db.{collection}.find({query_filter}).sort({{uid: 1}}).toArray()",
+        **run_mongo_op_kwargs,
+    )
+    assert find_documents.succeeded and len(find_documents.data) == 2
+
+    for index, test_doc in zip(range(len(o_test_docs)), o_test_docs):
+        db_doc = find_documents.data[index]
+
+        for key, val in test_doc.items():
+            assert db_doc[key] == val
+
+
+async def secondary_mongo_uris_with_sync_delay(ops_test: OpsTest, rs_status_data):
+    primary_optime_date = [
+        datetime.strptime(member["optimeDate"], "%Y-%m-%dT%H:%M:%S.%fZ")
+        for member in rs_status_data["members"]
+        if member["stateStr"].upper() == "PRIMARY"
+    ][0]
+
+    secondaries = []
+    for member in rs_status_data["members"]:
+        if member["stateStr"].upper() != "SECONDARY":
+            continue
+
+        unit_id = member["name"].split(".")[0].split("-")[-1]
+        member_optime_date = datetime.strptime(member["optimeDate"], "%Y-%m-%dT%H:%M:%S.%fZ")
+
+        host = await mongodb_uri(ops_test, [unit_id])
+
+        delay_seconds = (primary_optime_date - member_optime_date).total_seconds()
+
+        secondaries.append({"uri": host, "delay": math.fabs(delay_seconds)})
+
```

Comment:
this is so much of a nit that I debated commenting or not, so do whatever you want here
```suggestion
    secondaries = []
    for member in rs_status_data["members"]:
        if member["stateStr"].upper() != "SECONDARY":
            continue

        unit_id = member["name"].split(".")[0].split("-")[-1]
        member_optime_date = datetime.strptime(member["optimeDate"], "%Y-%m-%dT%H:%M:%S.%fZ")
        host = await mongodb_uri(ops_test, [unit_id])
        delay_seconds = (primary_optime_date - member_optime_date).total_seconds()
        secondaries.append({"uri": host, "delay": math.fabs(delay_seconds)})

```